source/common/upstream/load_balancer_impl.cc

```diff
@@ -562,6 +562,30 @@ HostConstSharedPtr LoadBalancerContextBase::selectOverrideHost(const HostMap* ho
   return nullptr;
 }
 
+LoadBalancerContext::OverrideHostStatus LoadBalancerContextBase::createOverrideHostStatus(
+    const std::vector<envoy::config::core::v3::HealthStatus>& statuses) {
+  LoadBalancerContext::OverrideHostStatus override_host_statuses = 0;
+  for (auto status : statuses) {
+    switch (status) {
+    case envoy::config::core::v3::HealthStatus::UNKNOWN:
+    case envoy::config::core::v3::HealthStatus::HEALTHY:
+      override_host_statuses |= HealthyStatus;
+      break;
+    case envoy::config::core::v3::HealthStatus::UNHEALTHY:
+    case envoy::config::core::v3::HealthStatus::DRAINING:
+    case envoy::config::core::v3::HealthStatus::TIMEOUT:
+      override_host_statuses |= UnhealthyStatus;
+      break;
+    case envoy::config::core::v3::HealthStatus::DEGRADED:
+      override_host_statuses |= DegradedStatus;
+      break;
+    default:
+      break;
+    }
```

Comment:
I think this mapping needs to be explained in the proto comments. Or perhaps the proto should be limited to specifying healthy/unhealthy/degraded.

Reply:
This mapping is explained in the proto comments of `HealthStatus`.

```
enum HealthStatus {
  // The health status is not known. This is interpreted by Envoy as *HEALTHY*.
  UNKNOWN = 0;

  // Healthy.
  HEALTHY = 1;

  // Unhealthy.
  UNHEALTHY = 2;

  // Connection draining in progress. E.g.,
  // `<https://aws.amazon.com/blogs/aws/elb-connection-draining-remove-instances-from-service-with-care/>`_
  // or
  // `<https://cloud.google.com/compute/docs/load-balancing/enabling-connection-draining>`_.
  // This is interpreted by Envoy as *UNHEALTHY*.
  DRAINING = 3;

  // Health check timed out. This is part of HDS and is interpreted by Envoy as
  // *UNHEALTHY*.
  TIMEOUT = 4;

  // Degraded.
  DEGRADED = 5;
}
```